packages/gatsby/src/commands/build-html.ts

```diff
@@ -92,6 +92,21 @@ const renderHTMLQueue = async (
   })
 }
 
+class BuildHTMLError extends Error {
+  codeFrame = ``
+  context?: {
+    path: string
+  }
+
+  constructor(error: Error) {
+    super(error.message)
+
+    Object.keys(error).forEach(key => {
```

Comment:
A bit late, but this is not copying `stack` property/value because `stack` is not enumarable, I think we want to use `getOwnPropertyNames` here? See snippet below for return values for `Object.keys()` and `Object.getOwnPropertyNames`:

```
keys [ 'name', 'context' ]
getOwnPropertyNames [ 'stack', 'message', 'name', 'context' ]
```

Effect this have is that SSR errors don't have real stack traces - instead they only show

```
  WebpackError: ReferenceError: window is not defined
  
  - build-html.js:111 doBuildPages
    [i25304]/[gatsby]/dist/commands/build-html.js:111:24
```

Without pointing to actual problem (because stack is newly created and not copied from original error)

Reply:
Ahhh yeah. Good call out. Gonna fix this

Reply:
There is bit more regressions that we introduced during TS conversions - I also found few in `prepare-stack-trace` - starting point was https://github.com/gatsbyjs/gatsby/blob/c9c3d68b589a1feba64d232c6b496968083e8e88/packages/gatsby-cli/src/reporter/prepare-stack-trace.ts#L15-L22 not using `stack` as well, but I think there might have been more.

The regression of either of those things is that when there is runtime error during SSR, the stack trace is all wrong:
 - for the problem I commented on here it means that stack will be gatsby insides saying that error was thrown in `build-html.js` 
 - for the prepare-stack-trace (after above is handled) - it means stack will be using `render-page.js`(which is bundle we generate for SSR) instead of actual source files

To showcase this you can put something like this in one of components render method and do `gatsby build`

```
const foo = null
foo.bar = `baz`
```

Expected result would be that stack trace points to the file with this code and not to either gatsby internals or to `render-page.js`

Reply:
And not very earth shattering statement - we should have e2e case for this (because it's not first time we regressed with showing stack trace for SSR runtime errors)